### ISSUE TITLE
Checkout button absent in cart page on homedepot.com - user cannot check out

### ISSUE THREAD

**Issue by jschinnerer** [brave/brave-browser#7252]
## Description 
"Checkout" button is not displayed in "cart" page on homedepot.com. User unable to proceed with checkout.

## Steps to Reproduce
1. Go to homedepot.com
2. Put any item in cart.
3. Go to cart to proceed with check out.

## Actual result:
No "checkout" button is displayed. User unable to proceed with checkout.
Same result with shields up or down.
Same result as anonymous or signed in user.

## Expected result:
"Checkout" button displayed, user able to proceed with checkout, as in other browsers.
Tested vs. Chromium: Version 78.0.3904.108 (Official Build) Built on Ubuntu , running on Ubuntu 18.04 (64-bit)
Tested vs. Firefox: 70.0.1 (64-bit) on Ubuntu 18.04
Screenshots attached for Brave, Chromium, Firefox.
![Brave-HDCart-no-checkout_2019-12-06_10-10-13](https://user-images.githubusercontent.com/1641535/70345758-9914a880-1811-11ea-8e5f-a9e96a89b3da.png)
![Chromium-HDCart-has-checkout_2019-12-06](https://user-images.githubusercontent.com/1641535/70345759-9914a880-1811-11ea-899c-e2db0667558d.png)
![FIrefox-HDCart-has-checkout_2019-12-06](https://user-images.githubusercontent.com/1641535/70345761-9914a880-1811-11ea-9563-0ca96597bb49.png)

## Reproduces how often: 
Easily reproduced (every time for me).

## Brave version (brave://version info)
Brave | 1.0.1 Chromium: 78.0.3904.108 (Official Build) (64-bit)
-- | --
Revision | 4b26898a39ee037623a72fcfb77279fce0e7d648-refs/branch-heads/3904@{#889}
OS | Linux
JavaScript | V8 7.8.279.23
Flash | (Disabled)
User Agent | Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.108 Safari/537.36

## Version/Channel Information:
I don't have any other systems to test on at this time.

- Can you reproduce this issue with the current release?
Yes, with current release as above on my Ubuntu 18.04 LTS system.

- Can you reproduce this issue with the beta channel?
Not able to test.
- Can you reproduce this issue with the dev channel?
Not able to test.
- Can you reproduce this issue with the nightly channel?
Not able to test.

## Other Additional Information:
- Does the issue resolve itself when disabling Brave Shields?
No, see above.
- Does the issue resolve itself when disabling Brave Rewards?
N/A - Brave Rewards not active on this installation.
- Is the issue reproducible on the latest version of Chrome? 
No, see above. Also not reproducible on Firefox.

## Miscellaneous Information:
I noticed when I started using Brave that homedepot.com has more, often far more, cross-site trackers than most any other site I use frequently. No idea if that is a factor here.
I've had very few instances so far where sites are broken due to "shields up" in Brave. In those few cases "shields down" has fixed it so far, but not in this case.

**Comment by rebron** [comment]
@bradleyrichter Have you run into this too? No checkout button in the shopping cart at homedepot even with shields down?

**Comment by ryanbr** [comment]
I've created a login and tested this, was able to get a `Checkout` button,  when I added an item. If you clear the cookies and re-login @jschinnerer does it help?

**Comment by jschinnerer** [comment]
Apologies for slow response...
Just regressed this. Cleared cookies first. No change for me. Same result, logged in or not (login is not required to complete purchase from homedepot.com so anonymous shopper should get checkout button in any case).

Current brave version:
Version 1.3.115 Chromium: 80.0.3987.87 (Official Build) (64-bit)
On same system as originally reportd.

**Comment by jschinnerer** [comment]
Additional information:
The checkout button is also missing on the cart overlay that appears when an item is added to the cart from any product page.

**Comment by ryanbr** [comment]
@jschinnerer 

From Brave 1.4.96. I get 2 checkouts (one on the right and bottom)

![checkout2](https://user-images.githubusercontent.com/1659004/76674514-2dfb1f00-6615-11ea-96e9-6eb21a3a7daa.png)

Did notice one finger printing issue, maybe `Allow all device recognition` might help.

**Comment by jschinnerer** [comment]
Screenshots from Brave Version 1.4.96 Chromium: 80.0.3987.132
Still no checkout button in normal laptop/desktop screen window view (first screenshot).
AND new information, didn't try this before - if I narrow window to first layout collapse point, I do get the upper checkout button but not the lower (second screenshot) in Brave. 
I get both buttons in Chromium.
I do see the checkout buttons as expected, any window width, on plain Chromium Version 80.0.3987.87
![Screenshot_2020-03-15_12-01-16](https://user-images.githubusercontent.com/1641535/76708631-97cb0580-66b5-11ea-9d62-0f031ebeb0ef.png)
![Screenshot_2020-03-15_12-02-22](https://user-images.githubusercontent.com/1641535/76708634-9bf72300-66b5-11ea-900c-a13b60314782.png)

Don't know what this means:
"Did notice one finger printing issue, maybe Allow all device recognition might help."
Doesn't sound like something the average Home Depot online shopper would know about either.

**Comment by bsclifton** [comment]
@jschinnerer apologies that this is still not resolved. I gave it a shot using anonymous guest login and it checkout worked as expected

Shields down should solve issues like this- but for this specific case, you've confirmed it doesn't. What @ryanbr was referring to is a setting inside the shields. Instead of toggling on/off, you could try turning everything off. Here's an example:
<img width="359" alt="Screen Shot 2020-03-29 at 10 25 26 PM" src="https://user-images.githubusercontent.com/4733304/77878093-546fab80-720c-11ea-8229-a47d2483c220.png">
(everything set as OFF and then drop downs have the `All ___ allowed` option picked)

If this doesn't make a difference, then I'm curious about if you have extensions installed. You can pull up your extension list in Brave by visiting brave://extensions. If you have any extensions, you can try disabling those to see if it makes a difference. I would compare to the extensions you have in Chrome by visiting chrome://extensions (in Chrome)

Please let us know - apologies no real progress has been made here

**Comment by mkelandis** [comment]
If it helps I landed here because I have this same problem in good old chrome (+ pihole networking). My LAN network DNS uses a pi-hole and I think that has something to do with it. When I switch over to google's DNS it works (8.8.8.8). To me, that is likely an issue with the homedepot site forcing traffic to a tracker before enabling the sign-in button. YMMV.

**Comment by jschinnerer** [comment]
This seems to be healed, for the cart view anyhow. Current version:
Version 1.9.72 Chromium: 81.0.4044.138 (Official Build) (64-bit)

I was about to regress this with your suggestion above, turning everything off manually instead of turning shields off.
I first went to homedepot.com, put something in cart, went to view cart, just to check default behavior.
Now I have checkout buttons, in both right and left panes, same as in plain chromium or firefox.

No checkout button in cart overlay that appears when something is added to cart - that part is same.
But the primary issue, no checkout button for anonymous or logged in cart view, seems to be fixed in the above  version.

**Comment by bsclifton** [comment]
@jschinnerer are you using any extensions? (you can view at brave://extensions). I'm curious if those are having any conflicts. You can also open the JavaScript console by pressing either F12 or Ctrl + Shift + I (I as in Inspect, not L) and going to `Console` to see if there are any errors / warnings showing

I tried in both Nightly (1.11.28) and latest release of Brave (1.9.72) and everything is working for me as expected... both the add to cart and also the checkout

Besides extensions, I'm curious if you're using a VPN or opening this in a Tor tab? That might affect this too. With a regular window you should be getting the exact same behavior as Chrome (but with some trackers blocked)

**Comment by jschinnerer** [comment]
Sorry, thought I had answered these before.
No VPN, no Tor. Typical residential internet service, cable modem -> router/wireless -> my device, no special proxies etc.
Until recently I had no extensions at all (up thru previous comment). I added Google Translate recently. Just removed it and regressed. Same behavior.

I would close this as resolved since my original bug I can no longer repro on my system. I get both checkout buttons when I go to cart.

My secondary issue in the comment thread, of no "checkout" button in overlay resulting from "add to cart," may be my mistake understanding default behavior.
It looks like there is a difference between add to cart overlay behavior, depending on where "add to cart" is clicked. If in page listing multiple items, button in overlay is "view cart," but if on a single item  detail page then I do get a "checkout" button.
